terraso_backend/apps/graphql/schema/sites.py

```diff
@@ -192,8 +192,11 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         site = cls.get_or_throw(Site, "id", kwargs["id"])
         if not user.has_perm(Site.get_perm("change"), site):
             raise cls.not_allowed(MutationTypes.UPDATE)
-        project_id = kwargs.pop("project_id", False)
+        if Site.SETTINGS_FIELDS.intersection(kwargs.keys()):
+            if not user.has_perm(Site.get_perm("change_settings"), site):
+                raise cls.not_allowed(MutationTypes.UPDATE)
```

Comment:
suggestion: add a comment to explain what is happening here. Especially with the intersection() call.

Reply:
done